packages/transit-vehicle-overlay/src/components/markers/ModeCircles/index.js

```diff
@@ -1,64 +1,28 @@
-import React from "react";
-import PropTypes from "prop-types";
-
-import { transitVehicleType } from "@opentripplanner/core-utils/lib/types";
-import RotatedMarker from "../RotatedMarker";
-import makeIcons from "./make-icons";
+import * as StyledCircle from "./styled";
 import * as utils from "../../../utils";
 
-/**
- * renders circular leaflet markers for the vehicles
- * at higher zoom levels, the icon used in the marker is based on the vehicles mode information
- */
-export default function ModeCircles(props) {
-  const { zoom, vehicle, children, color, highlightColor, isTracked } = props;
-  const { lat, lon, heading, routeType } = vehicle;
-  const icon = makeIcons(zoom, routeType, color, highlightColor, isTracked);
-
-  return (
-    <RotatedMarker
-      icon={icon}
-      position={[lat, lon]}
-      rotationAngle={heading}
-      rotationOrigin="center center"
-      onClick={() => props.onVehicleClicked(vehicle, isTracked)}
-      zIndexOffset={isTracked ? 1000 : 0}
-    >
-      {children}
-    </RotatedMarker>
+const Circle = utils.makeBasicVehicleShape(
+  StyledCircle.Shape,
+  StyledCircle.TrackedShape,
+  zoom => {
+    const midZoom = 12;
+    const midSize = 13.0;
+    const farSize = 7.0;
+    return zoom >= midZoom ? midSize : farSize;
+  }
+);
+
+const CircledVehicle = ({ color, highlightColor, isTracked, routeType }) => {
+  return utils.makeVehicleIcon(
+    StyledCircle,
+    routeType,
+    color,
+    highlightColor,
+    isTracked
   );
-}
-
-ModeCircles.propTypes = {
-  /** map zoom: is part of the props due to redrawing this layer on map zoom */
-  zoom: PropTypes.number,
-
-  /** vehicle record  - @see: core-utils/types/transitVehicleType */
-  vehicle: transitVehicleType.isRequired,
-
-  /** tracking boolean + colors all work to color the marker */
-  isTracked: PropTypes.bool,
-
-  /** fill color (#AABBCC format) for all (non-tracked) map vehicle markers */
-  color: PropTypes.string,
-
-  /** fill color of tracked vehicle */
-  highlightColor: PropTypes.string,
-
-  /** Callback fired when the vehicle marker is clicked (vehicle: object) => {} */
-  onVehicleClicked: PropTypes.func,
-
-  /** React children */
-  children: PropTypes.arrayOf(PropTypes.element)
 };
 
-ModeCircles.defaultProps = {
-  zoom: null,
-  isTracked: false,
-  color: "",
-  highlightColor: "",
-  onVehicleClicked: (vehicle, isTracked) => {
-    utils.linterIgnoreTheseProps(vehicle, isTracked);
-  },
-  children: null
+export default {
+  Circle: utils.makeRotatedMarker(Circle),
+  CircledVehicle: utils.makeRotatedMarker(CircledVehicle)
```

Comment:
I'd kind of prefer to have this wrapping done up above and then just export these items without any additional modifications. This export default could even be removed and the other items exported individually. Like so:

```js
export const Circle = utils.makeRotatedMarker(
  utils.makeBasicVehicleShape(
    ...
  )
)
```

Reply:
Addressed in https://github.com/opentripplanner/otp-ui/pull/159/commits/62601bebc9975e7d5316f4525d21aa201c1288db.